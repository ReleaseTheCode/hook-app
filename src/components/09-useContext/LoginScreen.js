import React,{ useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {
    const { setUser } = useContext(UserContext);
    const userObj = {
        id: 1,
        name: 'richter'
    } 

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr/>
            <button
                className='btn btn-primary'
                onClick={ ()=> setUser(userObj) }
            >
                Crear User
            </button>
        </div>
    )
}
