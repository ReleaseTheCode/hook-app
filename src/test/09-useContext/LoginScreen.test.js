import { mount } from 'enzyme';
import React from 'react';
import { LoginScreen } from '../../components/09-useContext/LoginScreen';
import { UserContext } from '../../components/09-useContext/UserContext';


describe('Test on <LoginScreen/>', () => {
    const setUser = jest.fn();
    const userObj = {
        id: 1,
        name: 'richter'
    };

    const wrapper = mount(
        <UserContext.Provider value={ { setUser }}>
            <LoginScreen/> 
        </UserContext.Provider>
    );
    
    test('should render', () => {
        expect( wrapper).toMatchSnapshot();
    });

    test('should execute setUser with args', () => { 
        wrapper.find('button').prop('onClick')();
        expect( setUser ).toHaveBeenCalledWith( userObj );
    });
});
