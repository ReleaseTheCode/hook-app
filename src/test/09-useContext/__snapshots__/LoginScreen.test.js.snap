// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <LoginScreen/> should render 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Crear User
  </button>
</div>
`;
